src/Forms/RequiredFields.php

```diff
@@ -104,16 +104,29 @@ public function php($data)
                 $fieldName = $fieldName->getName();
             } else {
                 $formField = $fields->dataFieldByName($fieldName);
+                if (is_null($formField)) {
+                    $formField = $fields->dataFieldByName($fieldName . 'ID');
+                }
             }
 
             // submitted data for file upload fields come back as an array
-            $value = isset($data[$fieldName]) ? $data[$fieldName] : null;
+            $value = null;
+            if (isset($data[$fieldName])) {
+                $value = $data[$fieldName];
+            } elseif (isset($data[$fieldName . 'ID'])) {
+                $value = $data[$fieldName . 'ID'];
+            }
```

Comment:
The comment is about file upload fields, but you're doing something here with the `ID` suffix which is clearly more about `has_one` relations. I think the comment was meant to be above the `is_array` check below? But that's also not only about file fields.

Please replace this comment with something about checking for `ID` suffixes. Please also only add the suffix for fields which implement `HasOneRelationFieldInterface`.

Reply:
Updated